src/kirby/components/modal/services/modal.controller.ts

```diff
@@ -29,7 +29,7 @@ export class ModalController implements IModalController {
     modalCloseEvent.then((data) => {
       this.forgetTopModal();
       if (onCloseModal) {
-        onCloseModal(data);
+        onCloseModal(typeof data === 'object' && 'data' in data ? data.data : data);
```

Comment:
Quite complex to read. Maybe some different naming or an explanation would help.
Of course, I happen to know the bug and why the fix is there... But still. :)

Reply:
agreed, i will add a comment to clarify :)

Reply:
Super :)